### PR TITLE
[opentelemetry-fluentd] Add new port

### PR DESCRIPTION
**Pull request by niande-xbox**
Add opentelemetry-cpp-contrib's fluentd exporter.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says. (_I'm using vcpkg_install_copyright_)
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by LilyWangLL** [comment]
There are the following warnings after building `opentelemetry-fluentd:x64-windows`, it is a `ONLY_STATIC_LIBRARY` port on windows, but it uses dynamic CRT.
```
-- Installing: E:/PRreview/opentele/packages/opentelemetry-fluentd_x64-windows/share/opentelemetry-fluentd/copyright
-- Performing post-build validation
warning: The following binaries should use the Dynamic Debug (/MDd) CRT.
    E:\PRreview\opentele\packages\opentelemetry-fluentd_x64-windows\debug\lib\opentelemetry_exporter_fluentd_logs.lib links with: Dynamic Release (/MD)
    E:\PRreview\opentele\packages\opentelemetry-fluentd_x64-windows\debug\lib\opentelemetry_exporter_fluentd_trace.lib links with: Dynamic Release (/MD)
To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
error: Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: E:\PRreview\opentele\ports\opentelemetry-fluentd\portfile.cmake
Stored binaries in 1 destinations in 1.3 s.
```

File: ports/opentelemetry-fluentd/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    REPO open-telemetry/opentelemetry-cpp-contrib
+    REF 20721e51235565020fc7fef2ba1aee92fc5744e8  # Maps to 2.0.0
+    HEAD_REF main
+    SHA512 5761eb75a16f558b6ca1abcbb38a1f2f33c02f045f54972ffdcdeb35ec12b340930b3c119f55602ff80772ca1de1fdfffd4f89d158779e33d0c92e95460c7bb1
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}/exporters/fluentd"
+    OPTIONS
+        -DCMAKE_BUILD_TYPE=Release
```
Comment (Neumann-A) [inline ports/opentelemetry-fluentd/portfile.cmake:16]: This is wrong.

File: ports/opentelemetry-fluentd/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE
```
Comment (Neumann-A) [inline ports/opentelemetry-fluentd/portfile.cmake:17]:
```suggestion
```

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Review by LilyWangLL — APPROVED** [review]

File: ports/opentelemetry-fluentd/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,24 @@
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+endif()
+
```
Comment (LilyWangLL) [inline ports/opentelemetry-fluentd/portfile.cmake:4]:
```suggestion
```
Since this port doesn't support windows, so here's codes could be removed.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/opentelemetry-fluentd/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}/exporters/fluentd"
+    OPTIONS
+        -DBUILD_TESTING=OFF
+        -DBUILD_EXAMPLE=OFF
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
```
Comment (LilyWangLL) [inline ports/opentelemetry-fluentd/portfile.cmake:21]: I saw this port exports target file, but it doesn't install it. Could you please also fix it?
https://github.com/open-telemetry/opentelemetry-cpp-contrib/blob/20721e51235565020fc7fef2ba1aee92fc5744e8/exporters/fluentd/CMakeLists.txt#L192-L196C4

**Comment by dg0yt** [comment]
> Added !windows in the supports - which should address this!

I can't see that  Windows is unsupported. AFAIU the problem was the portfile's overriding of the build type.

File: ports/opentelemetry-fluentd/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    REF 20721e51235565020fc7fef2ba1aee92fc5744e8  # Maps to 2.0.0
+    HEAD_REF main
+    SHA512 5761eb75a16f558b6ca1abcbb38a1f2f33c02f045f54972ffdcdeb35ec12b340930b3c119f55602ff80772ca1de1fdfffd4f89d158779e33d0c92e95460c7bb1
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}/exporters/fluentd"
+    OPTIONS
+        -DBUILD_TESTING=OFF
+        -DBUILD_EXAMPLE=OFF
```
Comment (LilyWangLL) [inline ports/opentelemetry-fluentd/portfile.cmake:17]: The `CMakeLists.txt` in `fluentd` subdirectory doesn't use option `BUILD_EXAMPLE`, it has code [`add_subdirectory(example)`](https://github.com/open-telemetry/opentelemetry-cpp-contrib/blob/49a42a3bd950d3359f0b3921c13f6adc7cfc793a/exporters/fluentd/CMakeLists.txt#L91), so this option is useless.
```
CMake Warning at installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:318 (message):
  The following variables are not used in CMakeLists.txt:

      BUILD_EXAMPLE

  Please recheck them and remove the unnecessary options from the
  `vcpkg_cmake_configure` call.
```
If you want disable build example, you need add a patch as below:
```
if(0)
add_subdirectory(example)
endif()
```
Or add option `BUILD_EXAMPLE`:
```
if(BUILD_EXAMPLE)
add_subdirectory(example)
endif()
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/opentelemetry-fluentd/portfile.cmake (hunk 4 above)
Comment (LilyWangLL) [reply ports/opentelemetry-fluentd/portfile.cmake:21]: I saw your PR https://github.com/open-telemetry/opentelemetry-cpp-contrib/pull/297 has been merged, however, this change has not been synchronized with this PR. 

I tested the usage locally, and it still has some other issues like below error. It need add `include(CMakeFindDependencyMacro)` and `find_dependency` in `opentelemetry-fluentd-config.cmake`.
```
1> [CMake] CMake Error at E:/PRreview/vcpkg/installed/x64-windows/share/opentelemetry-fluentd/opentelemetry-fluentd-target.cmake:60 (set_target_properties):
1> [CMake]   The link interface of target "opentelemetry-fluentd::trace" contains:
1> [CMake] 
1> [CMake]     opentelemetry-cpp::api
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
```

**Comment by LilyWangLL** [comment]
Please apply the following patch to `opentelemetry-fluentd`, which is a fix for the usage.
[fix-usage.patch](https://github.com/microsoft/vcpkg/files/12012470/fix-usage.patch)

**Comment by dg0yt** [comment]
Well, co

> > Please apply the following patch to `opentelemetry-fluentd`, which is a fix for the usage. [fix-usage.patch](https://github.com/microsoft/vcpkg/files/12012470/fix-usage.patch)
> 
> Looking into whether this breaks anything because I don't see any explicit setting of opentelemetry-fluentd_FOUND like they had before.

It is probably still better than before:
In config mode, `<Pkg>_FOUND` is initialized to TRUE by CMake. `find_dependency` will add a relevant interaction with it.
The variable checks it did before the path aren't really useful, at least in the vcpkg port.

**Review by LilyWangLL — APPROVED** [review]

**Comment by LilyWangLL** [comment]
Usage passed on x64-windows.

**Comment by LilyWangLL** [comment]
> @LilyWangLL - are you able to merge on my behalf?

Pinging @vicroms @BillyONeal for response, this PR is ready to merge now.

**Review by vicroms — APPROVED** [review]

**Comment by BillyONeal** [comment]
> [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.

I can't substantiate this. `opentelemetry-fluentd` doesn't appear on repology and I didn't get search hits either. Should this be a feature of `opentelemetry-cpp`, or of a new port `opentelemetry-contrib` instead?

**Comment by BillyONeal** [comment]
Upon further reflection I realize now that the name starts with opentelemetry which matches close enough the org name `open-telemetry` so I think this is fine.

Thanks for the new port!

**Pull request merged**